packages/react-components/src/components/MessageThread.tsx

```diff
@@ -909,21 +909,25 @@ export const MessageThreadWrapper = (props: MessageThreadProps): JSX.Element =>
   const [readCountForHoveredIndicator, setReadCountForHoveredIndicator] = useState<number | undefined>(undefined);
 
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
-  const [inlineAttachments, setInlineAttachments] = useState<Record<string, string>>({});
+  const [inlineAttachments, setInlineAttachments] = useState<Record<string, Record<string, string>>>({});
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
   const onFetchInlineAttachment = useCallback(
-    async (attachment: FileMetadata): Promise<void> => {
-      if (!onFetchAttachments || attachment.attachmentType !== 'inlineImage' || attachment.id in inlineAttachments) {
+    async (attachment: FileMetadata[], message: Message): Promise<void> => {
+      if (!onFetchAttachments) {
         return;
       }
 
-      setInlineAttachments((prev) => ({ ...prev, [attachment.id]: '' }));
+      const messageId = message.messageId;
```

Comment:
If all we need here is messageId, should we only pass in messageId instead of the whole message?